adobe_vipm/flows/utils/parameter.py

```
@@ -317,3 +321,35 @@ def get_retry_count(order: dict) -> str | None:
+    Args:
+        order: MPT order.
+
+    Returns:
+        Updated MPT order with visibility constraints applied.
+    """
+    agreement_type = get_ordering_parameter(order, Param.AGREEMENT_TYPE.value)
+    agreement_type_value = agreement_type.get("value", AgreementType.NEW.value)
+    market_segment = get_for_product(settings, "PRODUCT_SEGMENT", order["product"]["id"])
+    visible_params = list(AGREEMENT_VISIBLE_PARAMETERS.get(agreement_type_value, []))
+    visible_params.extend(AGREEMENT_VISIBLE_PARAMETERS.get(market_segment, []))
+    updated_order = copy.deepcopy(order)  # noqa: WPS204
```

Ruff (0.15.2)

[warning] Invalid rule code in `# noqa`: WPS204

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)